README.md

```diff
@@ -1 +1,118 @@
 # xtractmime
+
+`xtractmime` is a [BSD-licensed](https://opensource.org/licenses/BSD-3-Clause)
+Python 3.6+ implementation of the [MIME Sniffing
+Standard](https://mimesniff.spec.whatwg.org/).
+
+Install from [`PyPI`](https://pypi.python.org/pypi/xtractmime):
+
+```
+pip install xtractmime
+```
+
+---
+
+## Basic usage
+
+Below mentioned are some simple examples of using `xtractmime.extract_mime`:
+
+```python
+>>> from xtractmime import extract_mime
+>>> body = b"Sample text content"
+>>> extract_mime(body)
+b'text/plain'
+>>>
+```
+
+```python
+>>> from xtractmime import extract_mime
+>>> content_types = (b'text/html',)
+>>> body = b''
+>>> extract_mime(body, content_types=content_types)
+b'text/html'
+>>> 
+```
+
+Additional functionality to check if a mime type belongs to a specific mime type group using 
+methods mentioned in `xtractmime.mimegroups`
+
+```python
+>>> from xtractmime.mimegroups import *
+>>> mime_type = b'text/html'
+>>> is_html_mime_type(mime_type)
+True
+>>> is_image_mime_type(mime_type)
+False
+>>>
+```
+
+---
+
+## API Reference
+
+### function `xtractmime.extract_mime : Optional[bytes]`
+
+Return `mimetype` as a byte string if the byte sequence in the provided content is recognized by `xtractmime`
+else return `None`
+
+#### Parameters
+
+* `body: bytes`
+* `content_types: Optional[Tuple[bytes]], default = None`
+* `http_origin: bool, default = True`
+* `no_sniff: bool, default = False`
+* `extra_types: Optional[Tuple[Tuple[bytes, bytes, Optional[Set[bytes]], bytes], ...]], default = None`
+* `supported_types: Set[bytes], default = None`
```

Comment:
The function parameters should be before the description, since the description usually only makes sense after you know the accepted parameters.

I also suggest avoiding using a header for the parameters section. You can just precede them by “Parameters:”.